src/store/services.js

```diff
@@ -2,95 +2,102 @@ import MockData from '../data/movies.json';
-        method: isNewMovie ?  "POST" : "PUT",
-        headers: {
-          "Content-Type": "application/json"
-        },
-        body:  JSON.stringify(formatedData)
-      })
-    }
-    
-    _transformMovie(movie) {
-      return {
-        "id": movie.id,
-        "title": movie.title,
-        "tagline": movie.tagline || "123",
-        "vote_average": movie.vote_average || 0,
-        "vote_count": movie.vote_count || 0,
-        "release_date": movie.release_date || "",
-        "poster_path": movie.poster_path,
-        "overview": movie.overview || "",
-        "budget": movie.budget|| 0,
-        "revenue": movie.revenue || 0,
-        "genres": Array.isArray(movie.genres) ? movie.genres : movie.genres.split(''),
-        "runtime": +movie.runtime || 0
-    }
+    let search = searchVal ? `&searchBy=title&search=${searchVal}` : "";
+    const res = await fetch(`${this._apiBase}${url}?sortBy=${sorter}&sortOrder=desc&filter=${filter}${search}`);
+
+
+    if (!res.ok) {
+      throw new Error(`Could not fetch ${url}` +
+        `, received ${res.status}`)
     }
+    return await res.json();
+  }
+
+  async getAllMovies(sorter, filter, searchVal) {
+    const res = await this.getResource(`/movies`, sorter, filter, searchVal);
+    return res.data.map(this._transformMovie);
+  }
+
+  async getMovie(id) {
+    const movie = await this.getResource(`/movies/${id}/`);
+    return this._transformMovie(movie);
+  }
+
+  async searchMovie(searchVal) {
+    const params = {
+      search: searchVal,
+      searchBy: "title"
+    };
+    const movie = await fetch(`${this._apiBase}/movies?` + new URLSearchParams(params));
+    let res = await movie.json();
+    return res.data.map(this._transformMovie);
+  }
 
-    _transformNewMovie(movie) {
-      return {
-        "title": movie.title,
-        "tagline": movie.tagline || "123",
-        "vote_average": movie.vote_average || 0,
-        "vote_count": movie.vote_count || 0,
-        "release_date": movie.release_date || "",
-        "poster_path": movie.poster_path,
-        "overview": movie.overview || "",
-        "budget": movie.budget|| 0,
-        "revenue": movie.revenue || 0,
-        "genres": Array.isArray(movie.genres) ? movie.genres : movie.genres.split(''),
-        "runtime": +movie.runtime || 0
+  async deleteMovie(id) {
+    return await fetch(`${this._apiBase}/movies/${id}?`, {
+      method: 'DELETE',
+    })
+  }
+
+  async saveMovie(movieData, isNewMovie) {
+    const formatedData = isNewMovie ? this._transformNewMovie(movieData) : this._transformMovie(movieData);
+    return await fetch("http://localhost:4000/movies", {
+      method: isNewMovie ? "POST" : "PUT",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(formatedData)
+    })
+  }
+
+  _transformMovie(movie) {
+    return {
+      "id": movie.id,
+      "title": movie.title,
+      "tagline": movie.tagline || "123",
```

Comment:
Could you please explain, for what purpose there is 123 ? Why not just empty string?

Reply:
When I used it with empty string, I've received the error from backend: 
tagline cannot be empty string. Currently I've changed value "123" to " ".